controllers/secret/secret_controller.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package secret
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/vmware-samples/load-balancer-operator-for-kubernetes/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupWithManager adds this reconciler to a new controller then to the
+// provided manager.
+func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		// Watch ConfigMap resources.
+		For(&corev1.Secret{}).
+		Complete(r)
+}
+
+// SecretReconciler reconciles avi-credentials and avi-ca secrets
+type SecretReconciler struct {
+	client.Client
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	CredentialInited bool
+	CAInited         bool
+}
+
+// AKOO will reconcile default secrets avi-controller-credentials and avi-controller-ca
+// If customer uses their own secrets, AKOO has to be restarted when secrets are updated
+func (r *SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("Secret", req.NamespacedName)
+
+	secret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, req.NamespacedName, secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Secret not found, will not reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if secret.Name != v1alpha1.AviCredentialName && secret.Name != v1alpha1.AviCAName {
+		return ctrl.Result{}, nil
+	}
+
+	// Avoid restart during bootup
+	if r.CredentialInited && r.CAInited {
+		log.Info("avi secret " + req.Name + " updated. Shutdown ako-operator")
```

Comment:
this shutdown will trigger only if both CA and cred get modified in a ADC? Why that is the case?

Reply:
No. This shutdown is triggered if either one of default secrets (avi-controller-credentials/avi-controller-ca) is modified. Note, this controller is watching on corev1.Secret{}

Reply:
then should it be an `or` operator here?